### PR TITLE
Updated emacs plugin

### LINKED ISSUE #4221: emacs plugins with emacs >= 24.4
Hi,

Using emacs plugins with emacs >= 24.4 trigger an unexpected close whatever the command you are using (e, emacs, emacs -nw, ...).

### PR DESCRIPTION
**Pull request by soredake**
1. updated minimal version from 23 to 24
2. added new aliase that fix unexpected close with -nw option (fixes #4221)
3. removed some old aliases
4. fixed line that detects open frames, now it works

### CODE REVIEW

**Comment by mcornella** [comment]
Can you describe what this does in detail?

**Comment by mcornella** [comment]
Thanks! @hjpotter92 @emanresusername can you test this and comment on it?

File: plugins/emacs/emacs.plugin.zsh [hunk 1]
```diff
@@ -10,23 +10,22 @@
...
 
-if "$ZSH/tools/require_tool.sh" emacs 23 2>/dev/null ; then
+if "$ZSH/tools/require_tool.sh" emacs 24 2>/dev/null ; then
     export EMACS_PLUGIN_LAUNCHER="$ZSH/plugins/emacs/emacsclient.sh"
 
     # set EDITOR if not already defined.
     export EDITOR="${EDITOR:-${EMACS_PLUGIN_LAUNCHER}}"
 
     alias emacs="$EMACS_PLUGIN_LAUNCHER --no-wait"
+    alias temacs="$EMACS_PLUGIN_LAUNCHER"
```
Comment (hjpotter92) [inline plugins/emacs/emacs.plugin.zsh]: I think you meant to have a `-nw` option here as well.

**Comment by emanresusername** [comment]
i can't claim to be an emacsclient expert, but seems to be working great for me.
running `e whatever.file` with no frames open now does the expected thing (i.e. open and switch to a new frame visiting `whatever.file`. instead of before where it would just do . . . well . . . nothing :/) so 👍 for that!
maybe some comments for the usecases for the new aliases? it seems like they're trying to do different things (correctly i'm sure), it's just not apparent to me what each one is meant for?

File: plugins/emacs/emacs.plugin.zsh (hunk 1 above)
Comment (soredake) [inline plugins/emacs/emacs.plugin.zsh]: fixed

File: plugins/emacs/emacs.plugin.zsh [hunk 2]
```diff
@@ -10,23 +10,22 @@
...
     alias e=emacs
+    # open terminal emacsclient
+    alias te="$EMACS_PLUGIN_LAUNCHER -nw"
 
     # same than M-x eval but from outside Emacs.
     alias eeval="$EMACS_PLUGIN_LAUNCHER --eval"
     # create a new X frame
     alias eframe='emacsclient --alternate-editor "" --create-frame'
 
-    # to code all night long
```
Comment (hjpotter92) [inline plugins/emacs/emacs.plugin.zsh]: Also, why remove these aliases? I understand that zsh would autocorrect the mistakes :stuck_out_tongue:

Comment (emanresusername) [inline plugins/emacs/emacs.plugin.zsh]: if you're trying to encourage usage of [thefuck](https://github.com/nvbn/thefuck) instead
i approve

**Comment by emanresusername** [comment]
worked fine for me all day today 👍

**Comment by emanresusername** [comment]
@mcornella @hjpotter92  anything gating merging in this one?

**Comment by mcornella** [comment]
I was waiting for some input from @soredake, but your points don't seem to be that big of a deal. 

1. The deleted aliases thing should be solved with `setopt correct` which automatically guesses which command you meant if you mispell it.

2. The `te` alias runs emacs ignoring the window system, so it runs entirely on the terminal: [source](https://www.gnu.org/software/emacs/manual/html_node/emacs/Initial-Options.html#index-_002dnw).

**Comment by mcornella** [comment]
So I am merging this because of your approval. Thanks! :+1:

**Pull request merged**